int32 type
Pos3D pos

